File: common/src/MissingEtSelection.cpp
//----------------------------------------------------------------------------//
// PoTATo's a Top Analysis Tool
// Project: common
// File:    MissingEtSelection.cpp
// Author:  Joscha Knolle
// Date:    2018-04-22
//----------------------------------------------------------------------------//

#include "MissingEtSelection.h"

MissingEtSelection::MissingEtSelection(
    bool cutmet,
    double minmet,
    double maxmet
) :
    _cutMissingEt(cutmet),
    _minMet(minmet),
    _maxMet(maxmet)
{
}

void MissingEtSelection::initialize(Result* res) {
    res->branch("missingEt", &_met);
    if(!_cutMissingEt) res->branch("passedMissingEt", &_passedMissingEt);
}

float MissingEtSelection::execute(Event* ev) {
    // MET cuts
    _met = ev->missingEt()->Et();
    _passedMissingEt = (_minMet<0.0 || _met>_minMet) && (_maxMet<0.0 || _met<_maxMet);
    if(_cutMissingEt && !_passedMissingEt) return 0.0;

    return 1.0;
}

File: common/python/MissingEtSelection.py
##----------------------------------------------------------------------------##
## PoTATo's a Top Analysis Tool
## Project: common
## File:    MissingEtSelection.py
## Author:  Joscha Knolle
## Date:    2018-04-30
##----------------------------------------------------------------------------##

import ROOT

from Action import Action
from Field import Field
from SimpleObservable import SimpleObservable

class MissingEtSelection(Action):

    def __init__(self, **options):
        Action.__init__(self, **options)
        self.setOption(options, 'cutMissingEt', True)
        self.setOption(options, 'minMissingEt', -1.0)
        self.setOption(options, 'maxMissingEt', -1.0)

    def getAction(self):
        self.action = ROOT.MissingEtSelection(
            self.cutMissingEt, self.minMissingEt, self.maxMissingEt
        )
        return Action.getAction(self)
